### PR TITLE
Implement UnpublishPageAction and Admin API for un-publishing pages

### PR DESCRIPTION
**Pull request by Tijani-Dia**
See #7682.

* Do the tests still pass? Yes
* Does the code comply with the style guide? Yes
* For Python changes: Have you added tests to cover the new/fixed behaviour? Yes

### CODE REVIEW

File: wagtail/admin/api/actions/unpublish.py
Comment (kaedroho) [inline wagtail/admin/api/actions/unpublish.py:14]: https://github.com/wagtail/wagtail/pull/7728#discussion_r778103456

File: wagtail/admin/api/actions/unpublish.py
Comment (kaedroho) [inline wagtail/admin/api/actions/unpublish.py:13]: I think we should leave this out of the API, since it should only be set on pages that were scheduled to be unpublished and I'm not sure if we can validate that here.

**Review by kaedroho — CHANGES_REQUESTED** [review]
This looks good to me, nice work again!

Just think we should remove a couple of the options from the API for now, then this would be good to go!

File: wagtail/admin/api/actions/unpublish.py
Comment (kaedroho) [inline wagtail/admin/api/actions/unpublish.py:15]: I think for consistency, with copy, we should rename this to ``recursive``: https://github.com/wagtail/wagtail/blob/main/wagtail/admin/api/actions/copy.py#L19

File: wagtail/admin/api/actions/unpublish.py
Comment (Tijani-Dia) [reply wagtail/admin/api/actions/unpublish.py]: Alright!

**Comment by Tijani-Dia** [comment]
Thanks @kaedroho! I've addressed your comments.

**Review by kaedroho — APPROVED** [review]
Looks good to me now! Just needs merging main or rebasing and it'll be ready to merge

**Comment by Tijani-Dia** [comment]
I merged main branch into this one.

**Comment by lb-** [comment]
@Tijani-Dia I have just rebased this on main, now that the publish page action work is in, will merge into main once the checks have passed.

**Comment by lb-** [comment]
@kaedroho as per other comment, I have merged this in but have not included any release notes for this and other API work, will leave that to you.

@Tijani-Dia thanks for the epic solution here also.

**Pull request merged**
